balancer-js/src/types.ts

```diff
@@ -1,6 +1,48 @@
 import { BigNumberish } from '@ethersproject/bignumber';
 import { Network } from './constants/network';
 import { Contract } from '@ethersproject/contracts';
+import { PoolDataService, TokenPriceService } from '@balancer-labs/sor';
+import { Provider } from '@ethersproject/providers';
+
+export interface BalancerSdkConfig {
+    //use a known network or provide an entirely custom config
+    network: Network | BalancerNetworkConfig;
+    rpcUrl: string;
+    //default to JsonRpcProvider if not provided
+    provider?: Provider;
```

Comment:
I think we should avoid accepting ethers types as inputs as much as possible. There can often be TS errors thrown because of mismatching ethers versions in these cases, especially around providers. Since we are accepting a rpcUrl then the JsonRpcProvider should always be initialized within the SDK. The only thing we might need to accept as an input here might be a Signer. wdyt?

Reply:
Totally fine for me. We default require the `rpcUrl` and initialize the `JsonRpcProvider` within the SDK.

As for the signer I'd say we wait until we have a need for it. At the moment I'd say we leave any signing external to the SDK.

Reply:
Hey @garethfuller, I went ahead and merged these PRs into a single on pointing at the balancer labs repo https://github.com/balancer-labs/balancer-sdk/pull/22.

The change for the above is in this comment:

https://github.com/balancer-labs/balancer-sdk/pull/22/commits/3a10856a31688bbb970fe8401b7c8fa63e1b3b4c